modules/microshift-cni.adoc

```diff
@@ -0,0 +1,304 @@
+//====
+//The default CNI configuration is intended to match the developer environment described in MicroShift Development Environment on RHEL 8.
+//====
+//Q: What's the default production configuration?
+
+{product-title} uses OVN-Kubernetes, a virtualized network for pods and services. The OVN-Kubernetes Container Network Interface (CNI) plug-in is the network provider for the cluster. OVN-Kubernetes is based on Open Virtual Network (OVN) and provides an overlay-based networking implementation.
+
+A cluster that uses the OVN-Kubernetes network provider also runs Open vSwitch (OVS) on the node. OVN configures OVS on the node to implement the declared network configuration. OVS runs as a systemd service on the {product-title} node. The OVS RPM package is installed as a dependency to the `microshift-networking` RPM package. OVS is started immediately when the `microshift-networking` RPM is installed.
+
+By default, three performance optimizations are applied to OVS services to minimize resource consumption:
+
+* CPU affinity to `ovs-vswitchd.service` and `ovsdb-server.service`
+* `no-mlockall` to `openvswitch.service`
+* Limit handler and `revalidator` threads to `ovs-vswitchd.service`
+
+With {product-title}, each node is contained to a single machine and single {product-title}, so each deployment will require individual configuration, pods, and settings.
+
+== Network features
+Networking features available with {product-title} include:
+
+* Kubernetes network policy
+* Dynamic node IP
+* Secondary gateway interface
+* Dual stack
+//are these the correct features relevant to MicroShift 4.12?
+//https://docs.openshift.com/container-platform/4.11/networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.html#nw-ovn-kubernetes-matrix_about-ovn-kubernetes
+
+//Q: are there immutable network settings we should tell users about?
+== {product-title} networking components and services overview
+This brief overview describes networking components and their operation in {product-title}. The `microshift-networking` RPM is a package that pulls in any networking related dependencies and certain systemd services to initialize networking, for example, the `microshift-ovs-init` systemd service.
+
+NetworkManager::
+NetworkManager is required by OVN-Kubernetes to set up the initial gateway bridge on the {product-title} node. NetworkManager and NetworkManager-OVS RPM packages are installed as dependencies to the `microshift-networking` RPM package. NetworkManager is configured to use the `keyfile` plugin by default and restarts after installation of the `microshift-networking` RPM package.
+//Q: Meaning the microshift networking RPM replaces the keyfile plugin?
+microshift-ovs-init::
+The `microshift-ovs-init.service` is installed by the `microshift-networking` RPM package as a one-shot systemd service. The 'microshift-ovs-init.service` executes the `configure-ovs.sh` script which uses NetworkManager commands to set up the OVS gateway bridge.
```

Comment:
Does a user need to know this here? Maybe this level of detail can go into the troubleshooting section?

Reply:
@dhellmann Is this a likely error, or an outlier, do you think? We can definitely move to Troubleshooting; especially if this an automated step that could potentially go awry but can be resolved. What would be evidence of NetworkManager not picking up keyfile: is there a specific error we can reference? Would the resolution be to repeat the install of the `microshift-networking` RPM package? (All the RPMs go in during installation, is that correct?)